### ISSUE TITLE
Sync to Teams button not working

### ISSUE THREAD

**Issue by cwdata** [SharePoint/sp-dev-docs#6528]
## Category

- [x] Bug

## Describe the bug
I created a new personal Teams App using SharePoint Framework 1.11. I could install this app in several other tenants but not in our production tenant. 
The procedure is always the same: I upload the app to the tenant app catalog and check 'Make this solution available to all sites'. After that I open the Files ribbon and click on the button 'Sync to Teams'.
This button sends out a request to an API '/_api/web/tenantappcatalog/SyncSolutionToTeams'. But that API seems to be broken in our production tenant.
It replies with status 500 'Internal Server Error'. The response is:
`{"error":{"code":"-2146232832, Microsoft.SharePoint.SPException","message":{"lang":"de-DE","value":"Cannot find teams app to update"}}}`
The app is new to this tenant, so it is expected that the API does not find a previous version it could update.
In other tenants this API sends back a status of 200 OK when installing our app using the same package.
The user performing the above steps has the 'Global Admin' role.

## Steps to reproduce
Only happens in one tenant, but there the issue is persistent (tried for several days). Also deleted the app from the catalog and all recycle bins twice, re-added it and tried again without any change in the outcome.

## Expected behavior
We should be able to install this SPFx teams app to our production tenant.

## Environment details (*development & target environment*)
- **Your Developer Environment**: Windows 10
- **Target Environment**: SharePoint Online
- **Framework**: Node.js v10
- **Browser(s)**: Chrome, current version
- **Tooling**: VS Code, SPFx v1.11.0
- **Additional details**: Package installs perfectly fine in other tenants

### Additional context
Microsoft support refused to help because the problem is somehow development related...

**Comment by andrewconnell** [comment]
As this appears to be limited to one tenant, you're best bet is to open a ticket with MSFT support via the M365 portal.

**Comment by cwdata** [comment]
@andrewconnell I agree with you that this should be handled by MSFT support. Therefore I tried that first.
But as I've written they refused to help because it might be a little bit related to development and told me I should post to this forum.
The fact that the package works perfectly fine on other tenants has been completely ignored.
So that's the usual 'we are not in charge of this' ping pong - what could we do? I hope someone from Microsoft might read that here and might be interested in why the API fails on our tenant. Not being very confident in that this would happen though...

**Comment by rbarten** [comment]
@cwdata : I have the same error message I'm trying to sync the app from the SharePoint App Catalog to Teams. Did you ever solved this issue for your tenant?
I also I'm not able to remove the App from the Teams App Catalog, which was synced from the SharePoint App Catalog before, using the Teams Admin UI. 
I get the message:
"Teams App with id '{id}' was not found in the catalog" for call DELETE 'https://graph.microsoft.com/beta/appCatalogs/teamsApps/{id}'

**Comment by cwdata** [comment]
@rbarten,
as we never got any support on this issue from Microsoft we changed the id of our app. Then it could be synced to Teams on that particular tenant. We then replaced the app on all other tenants with the one that holds the new id.
We only have one version of the app in our catalog because the API failed to deployed it while it had its original id. So you might end up with two versions in your catalog if you don't find a way to delete the old one.
